charts/gardener/controlplane/charts/application/templates/rbac-user.yaml

```diff
@@ -83,6 +83,100 @@ aggregationRule:
       gardener.cloud/role: project-member
 rules: []
 
+# Cluster role granting viewer permissions for the resources in the gardener API group 
+---
+apiVersion: {{ include "rbacversion" . }}
+kind: ClusterRole
+metadata:
+  name: gardener.cloud:viewer
+  labels:
+    gardener.cloud/role: viewer
+rules:
+- apiGroups:
+  - core.gardener.cloud
+  - seedmanagement.gardener.cloud
+  - dashboard.gardener.cloud
+  - settings.gardener.cloud
+  - operations.gardener.cloud
+  resources:
+  - '*'
```

Comment:
Recently a new resource `ControllerDeployment` was introduced to hide some specs from some clients.
See https://github.com/gardener/gardener/pull/3995 for more details.

My point is that maybe it should not be allowed to read `controllerdeployments` just like it is not allowed for secrets.

`shootstates` also contain sensitive information that should not be made visible.


There are some other resources like `(cluster)openidconnectpresets`, `managedseeds`, `bastions`, `shootextensionstatuses` that needs to be also evaluated.

Reply:
With the assumption that such role will be granted to Gardener operators, i.e., people being part of the respective team operating a Gardener landscape, I would think the only resources that need to be excluded are:

- `controllerregistrations`
- `controllerdeployments`
- `shootstates`

as they might contain confidential/secret data.

Reply:
cc @rfranzke @timuthy @petersutter @mvladev

Reply:
@rfranzke View permissions removed for 
- controllerregistrations
- controllerdeployments
- shootstates